sphinx/util/fileutil.py

```diff
@@ -23,7 +23,7 @@ def _template_basename(filename: str | os.PathLike[str]) -> str | None:
     basename = os.path.basename(filename)
     if basename.lower().endswith('_t'):
         return str(filename)[:-2]
-    elif basename.lower().endswith(".jinja"):
```

Comment:
I'm not sure about the removal of `.lower()` here; what's the thinking behind that?

Reply:
@AA-Turner Although it might seem non-ideal in some ways to allow people to create template files named arbitrary combinations of `[Jj][Ii][Nn][Jj][Aa]`, initially I still feel that it's better to compare the file suffix case-insensitively.

Ref: https://www.hanselman.com/blog/git-is-casesensitive-and-your-filesystem-may-not-be-weird-folder-merging-on-windows for example.